### PR TITLE
[Compute] Upgrade api-version for VM and VMSS from '2020-12-01' to '2021-03-01'

### PR DESCRIPTION
**Pull request by zhoxing-ms**
**Description**

Upgrade api-version for VM and VMSS from `2020-12-01` to `2021-03-01`
Bump SDK `azure-mgmt-compute` from `20.0.0` to `21.0.0`

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

**Review by Juliehzl — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/export_features.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
   "Color": "Red",
   "Region": "West US",
-  "FeatureManagement": {
+  "feature_management": {
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/export_features.json:4]: Why?

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/export_features.json (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/appconfig/tests/latest/export_features.json:4]: This is automatically modified during the test running, not manually modified by me

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 2]
```diff
@@ -146,7 +146,7 @@ def default_api_version(self):
             'galleries': '2019-12-01',
             'gallery_images': '2020-09-30',
             'gallery_image_versions': '2020-09-30',
-            'virtual_machine_scale_sets': '2020-12-01'
+            'virtual_machine_scale_sets': '2021-03-01'
```
[the remark this answers is not in the document]
Comment (zhoxing-ms) [reply src/azure-cli-core/azure/cli/core/profiles/_shared.py:149]: OK, we may consider this in the future~

**Comment by andyliuliming** [comment]
+ @FumingZhang

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml [hunk 3]
```diff
@@ -121,63 +121,34 @@ interactions:
...
-        \   \"enableRBAC\": true,\n   \"networkProfile\": {\n    \"networkPlugin\"\
-        : \"kubenet\",\n    \"loadBalancerSku\": \"standard\",\n    \"loadBalancerProfile\"\
-        : {\n     \"managedOutboundIPs\": {\n      \"count\": 1\n     }\n    },\n\
-        \    \"podCidr\": \"10.244.0.0/16\",\n    \"serviceCidr\": \"10.0.0.0/16\"\
-        ,\n    \"dnsServiceIP\": \"10.0.0.10\",\n    \"dockerBridgeCidr\": \"172.17.0.1/16\"\
-        ,\n    \"outboundType\": \"loadBalancer\"\n   },\n   \"maxAgentPools\": 100\n\
-        \  },\n  \"sku\": {\n   \"name\": \"Basic\",\n   \"tier\": \"Free\"\n  }\n\
-        \ }"
+      string: "{\n  \"code\": \"ServicePrincipalNotFound\",\n  \"message\": \"Service
+        principal clientID: http://clitest000002 not found in Active Directory tenant
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml:140]: principal clientID: http://clitest000002 not found in Active Directory tenant
        54826b22-38d6-4fb2-bad9-b7b93a3e9c5a, Please see https://aka.ms/aks-sp-help
        for more details.\"\n }"

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml [hunk 4]
```diff
@@ -191,10 +162,23 @@ interactions:
       x-ms-ratelimit-remaining-subscription-writes:
       - '1199'
     status:
-      code: 201
-      message: Created
+      code: 400
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml:165]: why bad request.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml [hunk 5]
```diff
@@ -121,63 +121,34 @@ interactions:
...
       Content-Type:
       - application/json; charset=utf-8
       ParameterSetName:
       - --resource-group --name --location --dns-name-prefix --node-count --ssh-key-value
         --service-principal --client-secret --ppg
       User-Agent:
-      - python/3.6.10 (Linux-5.10.25-linuxkit-x86_64-with) msrest/0.6.21 msrest_azure/0.6.3
-        azure-mgmt-containerservice/11.2.0 Azure-SDK-For-Python AZURECLI/2.23.0 (DOCKER)
+      - python/3.8.0 (Windows-10-10.0.19041-SP0) msrest/0.6.21 msrest_azure/0.6.3
+        azure-mgmt-containerservice/11.0.0 Azure-SDK-For-Python AZURECLI/2.24.0
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml:132]: Please re-record the live test with latest acs module dependency. Currently, we are using azure-mgmt-containerservice/11.2.0 with api version 2021-03-01, however this recording takes azure-mgmt-containerservice/11.0.0 with api version 2020-12-01.

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml [hunk 6]
```diff
@@ -121,63 +121,34 @@ interactions:
...
-        \ \"nodeResourceGroup\": \"MC_clitest000001_cliakstest000001_westus2\",\n\
-        \   \"enableRBAC\": true,\n   \"networkProfile\": {\n    \"networkPlugin\"\
-        : \"kubenet\",\n    \"loadBalancerSku\": \"standard\",\n    \"loadBalancerProfile\"\
-        : {\n     \"managedOutboundIPs\": {\n      \"count\": 1\n     }\n    },\n\
-        \    \"podCidr\": \"10.244.0.0/16\",\n    \"serviceCidr\": \"10.0.0.0/16\"\
-        ,\n    \"dnsServiceIP\": \"10.0.0.10\",\n    \"dockerBridgeCidr\": \"172.17.0.1/16\"\
-        ,\n    \"outboundType\": \"loadBalancer\"\n   },\n   \"maxAgentPools\": 100\n\
-        \  },\n  \"sku\": {\n   \"name\": \"Basic\",\n   \"tier\": \"Free\"\n  }\n\
-        \ }"
+      string: "{\n  \"code\": \"ServicePrincipalNotFound\",\n  \"message\": \"Service
```
Comment (FumingZhang) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml:139]: Please re-record the live test with valid service principal id and secret. You can pass sp id and secret by environment variables. May refer to [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-testsdk/azure/cli/testsdk/preparers.py#L194-L195).

**Review by FumingZhang — CHANGES_REQUESTED** [review]
Please re-record the live test results for "test_aks_create_with_ppg" and "test_aks_create_with_ppg_msi".

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml (hunk 5 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml:132]: OK

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml (hunk 6 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml:139]: OK

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml (hunk 3 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml:140]: Good catch!

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml]: Fixed

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml (hunk 4 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_ppg.yaml]: This is the response which containing the error message 
> principal clientID: http://clitest000002 not found in Active Directory tenant

This has been solved by setting principal in the environment variable

**Review by FumingZhang — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

**Review by andyliuliming — APPROVED** [review]

**Pull request merged**